src/bin/driver.rs

```diff
@@ -168,14 +173,13 @@ async fn tree_init(client_0: &CollectorClient, client_1: &CollectorClient) -> io
 async fn add_keys(
     cfg: &config::Config,
     clients: (&CollectorClient, &CollectorClient),
-    all_keys: (&[vidpf::VidpfKey], &[vidpf::VidpfKey]),
+    all_keys: &[(VidpfKey, VidpfKey, String, Vec<Field128>)],
     all_proofs: (&[Vec<Field128>], &[Vec<Field128>]),
     all_nonces: &[[u8; 16]],
     all_jr_parts: &[[[u8; 16]; 2]],
     num_clients: usize,
     malicious_percentage: f32,
 ) -> io::Result<()> {
-    use rand::distributions::Distribution;
     let mut rng = rand::thread_rng();
     let zipf = zipf::ZipfDistribution::new(cfg.unique_buckets, cfg.zipf_exponent).unwrap();
```

Comment:
All occurrences of unique_buckets in this function can be replaced with `num_attributes` if we're in the attribute mode, right?

Reply:
Those are different actually: `unique_buckets` is a Zipf parameter; `num_attributes` is the number of samples we make from the Zipf distribuition.